api/cpp/esp-idf/slint/src/esp_slint.cpp

```diff
@@ -0,0 +1,174 @@
+
+        if (m_window) {
+
+            if (touch_handle && has_touch_events) {
+                uint16_t touchpad_x[1] = { 0 };
+                uint16_t touchpad_y[1] = { 0 };
+                uint8_t touchpad_cnt = 0;
+
+                /* Read touch controller data */
+                esp_lcd_touch_read_data(*touch_handle);
+
+                /* Get coordinates */
+                bool touchpad_pressed = esp_lcd_touch_get_coordinates(
+                        *touch_handle, touchpad_x, touchpad_y, NULL, &touchpad_cnt, 1);
+
+                if (touchpad_pressed && touchpad_cnt > 0) {
+                    // ESP_LOGI(TAG, "x: %i, y: %i", touchpad_x[0], touchpad_y[0]);
+                    last_touch_x = touchpad_x[0];
+                    last_touch_y = touchpad_y[0];
+                    m_window->window().dispatch_pointer_move_event(
+                            slint::LogicalPosition({ float(last_touch_x), float(last_touch_y) }));
+                    if (!touch_down) {
+                        m_window->window().dispatch_pointer_press_event(
+                                slint::LogicalPosition(
+                                        { float(last_touch_x), float(last_touch_y) }),
+                                slint::PointerEventButton::Left);
+                    }
+                    touch_down = true;
+                } else if (touch_down) {
+                    m_window->window().dispatch_pointer_release_event(
+                            slint::LogicalPosition({ float(last_touch_x), float(last_touch_y) }),
+                            slint::PointerEventButton::Left);
+                    m_window->window().dispatch_pointer_exit_event();
+                    touch_down = false;
+                }
+            }
+
+            if (std::exchange(m_window->needs_redraw, false)) {
+                auto region = m_window->m_renderer.render(buffer1, size.width);
+                auto o = region.bounding_box_origin();
+                auto s = region.bounding_box_size();
+                if (s.width > 0 && s.height > 0) {
+                    if (buffer2) {
+                        // Assuming that using double buffer means that the buffer comes from the
+                        // driver and we need to pass the exact pointer.
+                        // https://github.com/espressif/esp-idf/blob/53ff7d43dbff642d831a937b066ea0735a6aca24/components/esp_lcd/src/esp_lcd_panel_rgb.c#L681
+                        esp_lcd_panel_draw_bitmap(panel_handle, 0, 0, size.width, size.height,
+                                                  buffer1.data());
+                        std::swap(buffer1, buffer2.value());
+                        // FIXME: wait until swap finishes
+                    } else {
+                        for (int y = o.y; y < o.y + s.height; y++) {
+                            for (int x = o.x; x < o.x + s.width; x++) {
+                                // Swap endianess to big endian
```

Comment:
Perhaps this should be guarded with an ifdef for `BSP_LCD_BIGENDIAN` ?

Reply:
But i don't depend on the BSP in this component.

Reply:
Why do you need to depend on the bsp in this component to have access to this pre-processor macro?

AFAIK `sdkconfig.h` is generated globally and accessible at component build time. (in fact, I see components in esp-idf include that).

How else would you like to make this configurable? Or do you think we should only support big-endian driven screens?

Reply:
That is defined in `managed_components/espressif__esp-box/include/bsp/display.h`  not in sdkconfig.h

Reply:
Indeed, I mixed up that with Kconfig.

I still do think that we should make Slint for esp-idf work with little endian screens, so IMO we should have Kconfig options for the selected pixel format.

Or do you think we should just support only big-endian?

Reply:
I see the following options as acceptable:
 1. somehow configure with a flag when constructing the EspPlartform
 2. using Kconfig

I guess it depends who is supposed to change this option. And what's the most convenient.

For now, i'm tempted to just go with the current approach and polish later.